### DISCUSSION TITLE
'Requirement' of Query Parameters is ignored when using Models

### DISCUSSION THREAD

**Discussion by BassMastaCod** [fastapi/fastapi#12412] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Annotated, Optional

from fastapi import FastAPI, Query
from pydantic import BaseModel
from sqlmodel import SQLModel, Field

app = FastAPI()


class PydanticModel(BaseModel):
    a: str
    b: Optional[str]
    c: str


class PydanticSubModel(PydanticModel):
    d: Optional[str]
    e: str


@app.get("/PydanticModel")
def get_pydantic_model(query: Annotated[PydanticModel, Query()]):
    return query


@app.get("/PydanticSubModel")
def get_pydantic_sub_model(query: Annotated[PydanticSubModel, Query()]):
    return query


class Model(SQLModel, table=True):
    a: str = Field(primary_key=True)
    b: Optional[str]
    c: str


class SQLSubModel(Model):
    d: Optional[str]
    e: str


@app.get("/SQLModel")
def get_sql_model(query: Annotated[Model, Query()]):
    return query


@app.get("/SubSQLModel")
def get_sql_sub_model(query: Annotated[SQLSubModel, Query()]):
    return query
```

### Description

FastAPI completely ignores whether fields are optional in Query Parameter Models. When using a Pydantic Model, all parameters are required (as seen in Swagger below). When using a SQLModel, inherited fields are always optional while directly declared fields are always required. This only applies when `table=True`, otherwise SQLModels act the same way as Pydantic Models

![Swagger](https://github.com/user-attachments/assets/f03dab07-77e6-4ca0-aad2-2a32c96b9dfa)

I understand that some of this may be intentional behavior that I am unfamiliar with but at the very least, I would not expect table=True in SQLModel to affect how the model is interpreted by FastAPI.

This is vaguely related to https://github.com/fastapi/fastapi/discussions/12212 as I had to nest models in order to get all query parameters to show which resulted in erratic behavior.

### Operating System

Windows

### FastAPI Version

0.115.0

### Pydantic Version

2.9.2

### Python Version

3.12

### Additional Context

SQLModel version 0.0.22

**Comment by n8sty** [comment]
Try this, should fix your docs:

```
class PydanticModel(BaseModel):
    a: str
    b: Optional[str] = None
    c: str
```

notice the `None` as a default value.

**Reply by BassMastaCod** [reply]
Thank you for the help! Your suggestion fixes most of this but my main impediment was regarding the inherited SQLModel Fields which are still all optional:
See the last endpoint in the following image; **a**, **b**, and **c** should match the second-to-last endpoint, which shows **a** and **c** as required.
![image](https://github.com/user-attachments/assets/7ca3c3af-55f3-4ba2-bc62-509106891d97)
I tried both `b: Optional[str] = None` and `b: Optional[str] = Field(default=None)` which each gave the same result.

Once again, setting `table=False` causes it to function the same as the Pydantic Model.

**Comment by BassMastaCod** [comment]
Not ideal, but here is a workaround:

In addition to https://github.com/fastapi/fastapi/discussions/12412#discussioncomment-10885364, the SQLModel can be split into 2 with the second only being used to set `table=True`. Then the SubModel can extend the `table=False` model.
```Python
class TableFalseSQLModel(SQLModel):
    a: str = Field(primary_key=True)
    b: Optional[str] = Field(default=None)
    c: str


class TableTrueSQLModel(TableFalseSQLModel, table=True):
    pass


class SQLSubModel(TableFalseSQLModel):
    d: Optional[str] = None
    e: str
```
This is not a permanent solution as it cannot be used for existing libraries and other unmodifiable code. There are additional disadvantages but it may help in the short term.
